media_driver/linux/common/ddi/media_libva_util.cpp

```diff
@@ -405,6 +411,23 @@ VAStatus DdiMediaUtil_AllocateSurface(
     uint32_t    gmmPitch;
     uint32_t    gmmSize;
     uint32_t    gmmHeight;
+
+    gmmPitch    = (uint32_t)gmmResourceInfo->GetRenderPitch();
+    if( DdiMediaUtil_IsExternalSurface(mediaSurface) && ( mediaSurface->pSurfDesc->uiVaMemType == VA_SURFACE_ATTRIB_MEM_TYPE_USER_PTR ) )
+    {
+        uint32_t gmmHAlign;
+        gmmHAlign = gmmResourceInfo->GetHAlign();
+        if((gmmPitch < mediaSurface->pSurfDesc->uiPitches[0]) && (((mediaSurface->pSurfDesc->uiPitches[0]-gmmPitch)%gmmHAlign) == 0))
+        {
+            gmmResourceInfo->OverridePitch(mediaSurface->pSurfDesc->uiPitches[0]);
+        }
+        else
+        {
+           DDI_ASSERTMESSAGE("the pitch does not match the basic requirement.");
```

Comment:
This won't work for the case when user allocated pitch _matches_ the one from gmmlib:):). You need either use `<=` above and always call OverridePitch even if they match or explicitly avoid error with else `(gmmPitch != mediaSurface->pSurfDesc->uiPitches[0])`

Reply:
yes,